Add file arrays schema

p-galligan opened the pull request:
Adds a draft file arrays schema and fixtures.

p-galligan committed:
add files

Add schema for files array and update fixtures

p-galligan committed:
update files definition

Make title required and only allow string (not null as well)

p-galligan committed:
fix typo

Removes unnecessary comma in object json

p-galligan committed:
fix file array schema

Changes oneOf to anyOf for file arrays to allow both to exist at once.

helrond commented on schemas/base.json:
```diff
@@ -102,6 +102,47 @@
+				},
+				"download": {
+					"$id": "#/definitions/files/properties/download",
+					"type": ["string"],
```
It looks like there is as `uri` type - is that a more appropriate type to use here? https://json-schema.org/understanding-json-schema/reference/string#resource-identifiers

p-galligan replied:
Sorry, I must have missed that! I’ll update it with the additional fixture for multiple files in the array.

helrond replied:
Obvs we'd want to use that type for the manifest as well. And then let's make sure the type checking is working by tweaking the invalid fixtures.

helrond replied:
OK after further conversation it looks like this is format checking (rather than type checking) which is currently not enforced. We'll leave this as a string for now and if we need to introduce custom validation later we can do that.

p-galligan committed:
Tweak files schema

Tweaks files schema to use a URI format for manifest and download fields.

Updates fixtures.

p-galligan committed:
remove unnecessary fixtures

Removes unnecessary fixtures and brackets
